ArduPlane/Plane.cpp

```diff
@@ -1058,7 +1062,30 @@ void Plane::update_quicktune(void)
     quicktune.update(control_mode->supports_quicktune());
 }
 #endif
+#if MODE_AUTOLAND_ENABLED
+/*
+  In certain pilot controlled modes other than NAV_TAKEOFF or Mode TAKEOFF,takeoff direction is initialized after arm when sufficient altitude and ground speed is obtained, then captured takeoff direction + offset used as landing direction in AUTOLAND
+*/
+void Plane::check_takeoff_direction()
+{
+    if (takeoff_direction_check_completed || takeoff_state.initial_direction.initialized) {
```

Comment:
Do we need both `takeoff_direction_check_completed` and `takeoff_state.initial_direction.initialized`?

Reply:
no...fixed